test/functional/p2p_segwit.py

```diff
@@ -336,8 +336,7 @@ def test_unnecessary_witness_before_segwit_activation(self):
 
         # Verify the hash with witness differs from the txid
         # (otherwise our testing framework must be broken!)
-        tx.rehash()
-        assert tx.sha256 != tx.calc_sha256(with_witness=True)
+        assert tx.rehash() != tx.getwtxid()
```

Comment:
In commit "test: simplify (w)txid checks by avoiding .calc_sha256 calls" (a82829f37e1ed298b6c2b6d2859d4bea65fe3dcc)

It seems like there might be a test bug here, and this check will no longer fail even if the hash is the same because `rehash` returns an `int` while `getwtxid` returns a hex `str`

Reply:
re: https://github.com/bitcoin/bitcoin/pull/32050#discussion_r2015670851

> Note that both the `rehash` and `getwtxid` methods return a hex string.

Oops, sorry I misread the code. I saw `rehash` was setting the `sha256` member, and didn't realize it was actually returning a different member.